src/pointer_ops.rs

```diff
@@ -0,0 +1,312 @@
+// Copyright 2020 Amari Robinson
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+#[cfg(feature = "alloc")]
+use crate::alloc::boxed::Box;
+#[cfg(feature = "alloc")]
+use crate::alloc::rc::Rc;
+#[cfg(feature = "alloc")]
+use crate::alloc::sync::Arc;
+use crate::UnsafeRef;
+use core::marker::PhantomData;
+use core::mem::ManuallyDrop;
+use core::ops::Deref;
+
+/// Base trait for pointer conversion operations.
```

Comment:
This isn't actually a base trait since we don't have sub-traits of it. It's just a normal trait.